fineract-core/src/main/java/org/apache/fineract/portfolio/calendar/service/CalendarUtils.java

```diff
@@ -170,15 +170,15 @@ private static Collection<LocalDate> getRecurringDates(final Recur recur, final
                 numberOfDays);
     }
 
-    private static Collection<LocalDate> convertToLocalDateList(final DateList dates, final LocalDate seedDate,
+    static Collection<LocalDate> convertToLocalDateList(final DateList dates, final LocalDate seedDate,
             final PeriodFrequencyType frequencyType, boolean isSkippMeetingOnFirstDay, final Integer numberOfDays) {
         final Collection<LocalDate> recurringDates = new ArrayList<>();
 
-        for (@SuppressWarnings("rawtypes")
-        final Iterator iterator = dates.iterator(); iterator.hasNext();) {
-            final Date date = (Date) iterator.next();
-            recurringDates.add((LocalDate) adjustDate(LocalDate.ofInstant(date.toInstant(), DateUtils.getDateTimeZoneOfTenant()), seedDate,
-                    frequencyType));
+        for (final Date date : dates) {
+            LocalDateTime dateTimeInProperTz = getLocalDateTimeFromICal4JDate(date);
```

Comment:
It would be nice to find a way to move away from Calendar...and make it work with Java date / datetime...

Reply:
Not this time, it's a bigger job.